inc/dbmysql.class.php

```diff
@@ -261,130 +237,116 @@ function result($result, $i, $field) {
     *
-    * @param $result MySQL result handler
-    *
-    * @return result array
-   **/
+    * @param mysqli_result $result MySQL result handler
+    * @return string[]|null array results
+    */
    function fetch_array($result) {
-
       return $result->fetch_array();
    }
 
-
    /**
-    * Fetch row of the next row of a Mysql query
-    *
-    * @param $result MySQL result handler
+    * fetch_row() of the next row of a Mysql query
     *
-    * @return result row
-   **/
+    * @param mysqli_result $result MySQL result handler
+    * @return mixed|null result row
+    */
    function fetch_row($result) {
-
       return $result->fetch_row();
    }
 
-
    /**
-    * Fetch assoc of the next row of a Mysql query
-    *
-    * @param $result MySQL result handler
+    * fetch_assoc() of the next row of a Mysql query
     *
-    * @return result associative array
-   **/
+    * @param mysqli_result $result MySQL result handler
+    * @return string[]|null result associative array
+    */
    function fetch_assoc($result) {
-
       return $result->fetch_assoc();
    }
 
+   /**
+    * Fetch object of the next row of an SQL query
+    * @param mysqli_result $result MySQL result handler
+    * @return object|null
+    */
+   function fetch_object($result) {
+      return $result->fetch_object();
+   }
 
    /**
     * Move current pointer of a Mysql result to the specific row
     *
-    * @param $result    MySQL result handler
-    * @param $num       row to move current pointer
-    *
+    * @param mysqli_result $result    MySQL result handler
+    * @param int $num       row to move current pointer
     * @return boolean
-   **/
+    */
    function data_seek($result, $num) {
       return $result->data_seek($num);
    }
 
-
    /**
-    * Give ID of the last insert item by Mysql
-    *
-    * @return item ID
-   **/
+    * Give ID of the last inserted item by Mysql
+    * @return mixed
+    */
    function insert_id() {
       return $this->dbh->insert_id;
    }
 
-
    /**
     * Give number of fields of a Mysql result
     *
-    * @param $result MySQL result handler
-    *
-    * @return number of fields
-   **/
+    * @param mysqli_result $result MySQL result handler
+    * @return int number of fields
+    */
    function num_fields($result) {
       return $result->field_count;
    }
 
-
    /**
     * Give name of a field of a Mysql result
     *
-    * @param $result  MySQL result handler
-    * @param $nb     number of columns of the field
-    *
-    * @return name of the field
-   **/
+    * @param mysqli_result $result  MySQL result handler
+    * @param int $nb     ID of the field
+    * @return string name of the field
+    */
    function field_name($result, $nb) {
-
       $finfo = $result->fetch_fields();
       return $finfo[$nb]->name;
    }
 
-
    /**
     * Get flags of a field of a mysql result
     *
-    * @param $result    MySQL result handler
+    * @todo BUGGY FUNCTION : param $field isn't used...
```

Comment:
This method is not called from GLPI's core; maybe just mark it as deprecated (maybe check in history if it has been used once; to know what replace it now).